com.unity.multiplayer.mlapi/Runtime/Core/NetworkedBehaviour.cs

```diff
@@ -46,103 +48,56 @@ internal enum NExec
         // RuntimeAccessModifiersILPP will make this `protected`
         internal BitWriter BeginSendServerRpc(ServerRpcSendParams sendParams, bool isReliable)
         {
-            // @mfatihmar (Unity) Begin: Temporary, placeholder implementation
-            var stream = new BitStream();
-            var writer = new BitWriter(stream);
+            var rpcQueueMananger = NetworkingManager.Singleton.RpcQueueManager;
+            if (rpcQueueMananger == null)
+            {
+                return null;
+            }
 
+            var writer = rpcQueueMananger.BeginAddQueueItemToOutboundFrame(RPCQueueManager.QueueItemType.ServerRpc, Time.realtimeSinceStartup, StandardRpc_ChannelName, 0, NetworkingManager.Singleton.ServerClientId, null);
             writer.WriteBit(false); // Encrypted
             writer.WriteBit(false); // Authenticated
             writer.WriteBits(MLAPIConstants.MLAPI_SERVER_RPC, 6); // MessageType
             writer.WriteUInt64Packed(NetworkId); // NetworkObjectId
             writer.WriteUInt16Packed(GetBehaviourId()); // NetworkBehaviourId
-
             return writer;
-            // @mfatihmar (Unity) End: Temporary, placeholder implementation
         }
 
         // RuntimeAccessModifiersILPP will make this `protected`
         internal void EndSendServerRpc(BitWriter writer, ServerRpcSendParams sendParams, bool isReliable)
         {
-            // @mfatihmar (Unity) Begin: Temporary, placeholder implementation
             if (writer == null) return;
 
-            var stream = (BitStream)writer.GetStream();
-            if (stream != null)
-            {
-                stream.PadStream();
-
-                var networkManager = NetworkingManager.Singleton;
-                if (ReferenceEquals(networkManager, null)) return;
-
-                var payload = new ArraySegment<byte>(stream.GetBuffer(), 0, (int)stream.Length);
-                // @mfatihmar (Unity) Begin: Temporary, inbound RPC queue will replace this workaround
-                if (networkManager.IsHost)
-                {
-                    networkManager.__loopbackRpcQueue.Enqueue((payload, Time.realtimeSinceStartup));
-                }
-                // @mfatihmar (Unity) End: Temporary, inbound RPC queue will replace this workaround
-                else
-                {
-                    networkManager.NetworkConfig.NetworkTransport.Send(networkManager.ServerClientId, payload, "STDRPC");
-                }
-            }
-
-            writer.SetStream(null);
-            // @mfatihmar (Unity) End: Temporary, placeholder implementation
+            var rpcQueueMananger = NetworkingManager.Singleton.RpcQueueManager;
+            rpcQueueMananger?.EndAddQueueItemToOutboundFrame(writer);
```

Comment:
we probably don't need to check `null` here too (similar to above).

Reply:
Ditto in regards to adding this to the new Jira task